WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/BaseStatsUseCase.kt

```diff
@@ -200,6 +191,43 @@ abstract class BaseStatsUseCase<DOMAIN_MODEL, UI_STATE>(
         return buildLoadingItem() + listOf(BlockListItem.Empty(textResource = R.string.stats_no_data_yet))
     }
 
+    private fun updateState() {
+        updateJob?.let { job ->
+            if (job.isActive) {
+                job.cancel()
+            }
+        }
+        updateJob = GlobalScope.launch(backgroundDispatcher) {
```

Comment:
Is the `GlobalScope` necessary here? Wouldn't it be possible to pass down a scope from the ViewModel?

Reply:
I'll look into it, it's a great suggestion 👍

Reply:
I've changed the `BaseStatsUseCase` and the `BaseListUseCase` to implement CoroutineScope and use that instead. Their life cycle is longer than `ViewModel` so I'm not cancelling the jobs now. I can look into that in another PR.